### DISCUSSION TITLE
Developer build startup error

### DISCUSSION THREAD

**Discussion by alexmlt1971** [iina/iina#4499] [General]
I am download the latest nightly build of IINA. 
At startup, the system gives an error - the application cannot be checked.
What does a normal user (not a developer) need to do to use developer builds.

**Comment by yuxiqian** [comment]
Try right-click `IINA.app` and select "Open" to forcefully start it. That will override system security settings and might be risky.

Note that according to IINA developers, nightly version "is not recommended to be used as daily drive".

**Reply by alexmlt1971** [reply]
Understood.

**Reply by low-batt** [reply]
The answer from @yuxiqian is correct. This has to do with Apple's [code signing](https://developer.apple.com/support/code-signing/) system. From [Wikipedia](https://en.wikipedia.org/wiki/Code_signing):
> **Code signing** is the process of digitally signing executables and scripts to confirm the software author and guarantee that the code has not been altered or corrupted since it was signed

This security system attempts to stop hackers from patching a trojan into an executable file. If an executable file is signed macOS will check that it has not been altered. If that check fails macOS will refuse to execute the file. I think macOS will report something like "Code Signature Invalid".

Official releases of IINA are code signed. [Nightly builds](https://nightly.iina.io/) are unsigned, so macOS is making sure you really want to run an unsigned executable.

Nightly builds are generated from the `develop` branch. This is the code IINA developers are running. Code in the `develop` branch went through a review process and should be good, but has had significantly less testing than an official release. This and the lack of code signing is behind the caution about using a nightly build for your "daily drive".

**Comment by alexmlt1971** [comment]
After the release of the lastest stable build, there were a lot of small fixes in nightly builds. There is information when these changes will be formalized in a stable release?

**Reply by low-batt** [reply]
IINA does not officially publish future release plans. All contributors to IINA are volunteers. That means for many work/school priorities can delay work on IINA.

Currently it is looking like there will be another release soon to correct regressions being reported against IINA 1.3.2. This will include correcting lost localizations and disabling of VP9 hardware acceleration on Intel based Macs. Not clear yet what else might be included. If you encounter something that is working in IINA 1.3.1 and is malfunctioning in 1.3.2, please report it ASAP.

The recent releases were intended primarily to be bug fix releases. Some enhancements ended up being included. The current plan is for the next release (after the above mentioned regression fix release) to be a feature release. The primary goal of this release is to officially release the new plugin system. That system is in the release version of IINA, but is disabled as there are still problems that need to be addressed and documentation is needed.

Source control systems such as GitHub allow a development team to "freeze" sources for a release on a separate branch while continuing development work. In the case of IINA 1.3.2 the release was delayed due to a critical problem being discovered in the upgraded dependencies. So more changes have accumulated in the `develop` branch than usually would be the case after a release.

When making decisions about a release, open IINA [Pull requests](https://github.com/iina/iina/pulls) is another significant factor. A [pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests) is a request from a developer to "pull" code changes into IINA. This mechanism supports a review process where other developers can look at the proposed code changes and approve them or ask for changes. Once the changes are approved they are [merged](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/merging-a-pull-request) into the official IINA sources. As you can see there currently are lots of interesting changes waiting to be reviewed.

**Reply by alexmlt1971** [reply]
Thanks for the answer. It was precisely about offensive errors in localization, which were not in earlier releases.

**Reply by low-batt** [reply]
Yes, that is considered a must fix. We are currently waiting to hear from users about some new issues that seem to be regressions, but we are having trouble reproducing.
